### PR TITLE
refactor(api): project keys crypto

### PR DESCRIPTION
**Pull request by fsamin**
Signed-off-by: francois  samin <francois.samin@corp.ovh.com>

### CODE REVIEW

File: engine/api/migrate/refactor_proj_keys.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+	if err := rows.Close(); err != nil {
+		return sdk.WithStack(err)
+	}
+
+	var mError = new(sdk.MultiError)
+	for _, id := range ids {
+		if err := refactorProjectKeys(ctx, db, id); err != nil {
+			mError.Append(err)
+			log.Error(ctx, "migrate.RefactorProjectKeys> unable to migrate application_key %d: %v", id, err)
```
Comment (richardlt) [inline engine/api/migrate/refactor_proj_keys.go]:
```suggestion
			log.Error(ctx, "migrate.RefactorProjectKeys> unable to migrate project_key %d: %v", id, err)
```

File: engine/api/migrate/refactor_proj_keys.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
+package migrate
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/ovh/cds/engine/api/project"
```
Comment (sguiheux) [inline engine/api/migrate/refactor_proj_keys.go]: wrong place

File: engine/api/migrate/refactor_proj_keys.go [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+	"github.com/ovh/cds/engine/api/secret"
+	"github.com/ovh/cds/sdk"
+	"github.com/ovh/cds/sdk/log"
+
+	"github.com/go-gorp/gorp"
```
Comment (sguiheux) [inline engine/api/migrate/refactor_proj_keys.go]: wrong place

File: engine/api/project/dao_key.go [hunk 4]
```diff
@@ -2,99 +2,114 @@ package project
...
-// LoadAllKeys load all keys for the given project
-func LoadAllKeys(db gorp.SqlExecutor, proj *sdk.Project) error {
-	var res []dbProjectKey
-	if _, err := db.Select(&res, "SELECT * FROM project_key WHERE project_id = $1 and builtin = false", proj.ID); err != nil {
-		if err == sql.ErrNoRows {
-			return nil
-		}
-		return sdk.WrapError(err, "Cannot load keys")
-	}
+// LoadAllKeys load all keys for the given application
```
Comment (richardlt) [inline engine/api/project/dao_key.go]:
```suggestion
// LoadAllKeys load all keys for the given project.
```

File: engine/api/project/dao_key.go [hunk 5]
```diff
@@ -103,22 +118,30 @@ func DeleteProjectKey(db gorp.SqlExecutor, projectID int64, keyName string) erro
...
 
-func loadBuildinKey(db gorp.SqlExecutor, projectID int64) (sdk.ProjectKey, error) {
-	var k sdk.ProjectKey
-	var res dbProjectKey
-	if err := db.SelectOne(&res, "SELECT * FROM project_key WHERE project_id = $1 and builtin = true and name = 'builtin'", projectID); err != nil {
-		if err == sql.ErrNoRows {
-			return k, sdk.ErrBuiltinKeyNotFound
-		}
-		return k, sdk.WrapError(err, "Cannot load keys")
+func loadBuildinKey(db gorp.SqlExecutor, projectID int64) (*sdk.ProjectKey, error) {
```
Comment (richardlt) [inline engine/api/project/dao_key.go]:
```suggestion
func loadBuiltinKey(db gorp.SqlExecutor, projectID int64) (*sdk.ProjectKey, error) {
```

File: engine/api/project/dao_key.go [hunk 6]
```diff
@@ -103,22 +118,30 @@ func DeleteProjectKey(db gorp.SqlExecutor, projectID int64, keyName string) erro
...
+	query := gorpmapping.NewQuery(`
+	SELECT * 
+	FROM project_key
+	WHERE id = $1 
+	AND name = $2
```
Comment (richardlt) [inline engine/api/project/dao_key.go]: Maybe to remove

Comment (yesnault) [inline engine/api/project/dao_key.go]: to delete name=$2

File: engine/sql/187_project_key.sql [hunk 7]
```diff
@@ -0,0 +1,24 @@
+-- +migrate Up
+
+CREATE TABLE project_key_tmp AS SELECT * FROM project_key;
+ ALTER TABLE project_key_tmp ADD PRIMARY KEY (id);
```
Comment (richardlt) [inline engine/sql/187_project_key.sql]:
```suggestion
ALTER TABLE project_key_tmp ADD PRIMARY KEY (id);
```

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#12604.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✘  
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Pull request merged**